Framework/Algorithms/src/FilterByLogValue.cpp

```diff
@@ -134,34 +136,37 @@ void FilterByLogValue::exec() {
       std::vector<DateAndTime> times = log->timesAsVector();
       std::vector<DateAndTime>::iterator it;
       for (it = times.begin(); it != times.end(); ++it) {
-        SplittingInterval interval(lastTime, *it - tolerance, 0);
+        if (lastTime < *it - tolerance)
+          roi->addROI(lastTime, (*it - tolerance));
         // Leave a gap +- tolerance
         lastTime = (*it + tolerance);
-        splitter.emplace_back(interval);
       }
       // And the last one
-      splitter.emplace_back(lastTime, run_stop, 0);
+      if (lastTime < run_stop)
+        roi->addROI(lastTime, run_stop);
 
     } else {
       // ----- Filter by value ------
 
       // This function creates the splitter vector we will use to filter out
       // stuff.
-      const std::string logBoundary(this->getPropertyValue("LogBoundary"));
-      log->makeFilterByValue(splitter, min, max, tolerance, (logBoundary == CENTRE));
-
-      if (log->realSize() >= 1 && handle_edge_values) {
-        log->expandFilterToRange(splitter, min, max, TimeInterval(run_start, run_stop));
+      const TimeROI *tempTimeROI = &inputWS->run().getTimeROI();
+      bool centre = this->getPropertyValue("LogBoundary") == CENTRE;
+      if (log->realSize() > 0 && handle_edge_values) {
+        *roi =
+            log->makeFilterByValue(min, max, true, TimeInterval(run_start, run_stop), tolerance, centre, tempTimeROI);
+      } else {
+        *roi = log->makeFilterByValue(min, max, false, TimeInterval(0, 1), tolerance, centre, tempTimeROI);
       }
     } // (filter by value)
   }
 
-  g_log.information() << splitter.size() << " entries in the filter.\n";
+  g_log.information() << roi->numBoundaries() / 2 << " entries in the filter.\n";
```

Comment:
TimeROI should have a method for number of regions. I created [EWM1161](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=1161) to do that later.

Reply:
I updated this comment to make it more clear